go/libraries/doltcore/sqle/types.go

```diff
@@ -80,45 +81,135 @@ func nomsValToSqlVal(val types.Value) interface{} {
 	}
 }
 
-func SqlValToNomsVal(val interface{}) types.Value {
+func SqlValToNomsVal(val interface{}, kind types.NomsKind) (types.Value, error) {
 	if val == nil {
-		return nil
+		return nil, nil
 	}
 
-	switch e := val.(type) {
-	case bool:
-		return types.Bool(e)
-	case int:
-		return types.Int(e)
-	case int8:
-		return types.Int(e)
-	case int16:
-		return types.Int(e)
-	case int32:
-		return types.Int(e)
-	case int64:
-		return types.Int(e)
-	case uint:
-		return types.Uint(e)
-	case uint8:
-		return types.Uint(e)
-	case uint16:
-		return types.Uint(e)
-	case uint32:
-		return types.Uint(e)
-	case uint64:
-		return types.Uint(e)
-	case float32:
-		return types.Float(e)
-	case float64:
-		return types.Float(e)
-	case string:
+	switch kind {
+	case types.BoolKind:
```

Comment:
So go-mysql-server has logic for doing this kind of conversion which you should copy or use. It would be weird for us to evaluate boolean values differently on insert than in other contexts. Check it out: 

https://github.com/src-d/go-mysql-server/blob/master/sql/type.go#L698

The same logic applies to all these things: you should use the go-mysql-server logic to get as close to the dolt type as possible, then apply any dolt-specific logic (like unsigned, uuid, etc)

Reply:
Ohhh I wasn't aware of this, I can make the changes.